src/widgetastic_patternfly/__init__.py

```diff
@@ -1808,14 +1808,17 @@ class Modal(View):
 
     def __init__(self, parent, id=None, logger=None):
         self.id = id
-        if id:
-            self.ROOT = ParametrizedLocator(
-                './/div[normalize-space(@id)={@id|quote} and '
-                'contains(@class, "modal") and contains(@class, "fade") '
-                'and @role="dialog"]')
-
         View.__init__(self, parent, logger=logger)
 
+    def __locator__(self):
```

Comment:
In `View` or `Widget` when we set `ROOT` locator should auto-resolve.

I think it will be great if we fallow an approach like 
```
ROOT = ParametrizedLocator("{@locator}")
def __init__(self, parent, id=None, logger=None):
    self.id = id
    if id:
        self.locator = <some_loator_with id>
    else:
        self.locator = <default>
```

Reply:
I think implementing `__locator__` here is okay, though the pattern @digitronik suggested above does it without implementing the dunder method.

Why its broken in the first place, and why this fix works, is because of the WidgetMetaclass behavior in `__new__`.

Before the WidgetDescriptor instance is created when the unit test is defining the test view class, WidgetMetaclass is setting `__locator__` based on the presence of `ROOT` in the kwargs.  After this, when the WidgetDescriptor resolves into a View instance in the presence of a browser, `self.ROOT` is modified - but `self.__locator__` has already been created automatically based off of the original definition.